{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Retrieve study data from Metabolomics Workbench using REST API"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import Python modules..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:21.780228600Z",
     "start_time": "2023-10-09T13:19:21.682219700Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python: 3.11.5\n",
      "IPython: 8.15.0\n",
      "\n",
      "Mon Oct  9 15:19:21 2023\n"
     ]
    }
   ],
   "source": [
    "from __future__ import print_function\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import time\n",
    "import re\n",
    "\n",
    "import requests\n",
    "\n",
    "from IPython import __version__ as ipyVersion\n",
    "\n",
    "print(\"Python: %s.%s.%s\" % sys.version_info[:3])\n",
    "print(\"IPython: %s\" % ipyVersion)\n",
    "\n",
    "print()\n",
    "print(time.asctime())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The URL PATH**\n",
    "\n",
    "The MW REST URL consists of three main parts, separated by forward slashes, after the common prefix specifying the invariant base URL (https://www.metabolomicsworkbench.org/rest/):\n",
    "\n",
    "https://www.metabolomicsworkbench.org/rest/context/input_specification/output_specification\n",
    "    \n",
    "Part 1: The context determines the type of data to be accessed from the Metabolomics Workbench, such as metadata or results related to the submitted studies, data from metabolites, genes/proteins and analytical chemistry databases as well as other services related to mass spectrometry and metabolite identification:\n",
    "\n",
    "context = study | compound | refmet | gene | protein | moverz | exactmass\n",
    "\n",
    "Part 2: The input specification consists of two required parameters describing the REST request:\n",
    "\n",
    "input_specification = input_item/input_value\n",
    "\n",
    "Part 3: The output specification consists of two parameters describing the output generated by the REST request:\n",
    "\n",
    "output_specification = output_item/(output_format)\n",
    "\n",
    "The first parameter is required in most cases. The second parameter is optional. The input and output specifications are context sensitive. The context determines the values allowed for the remaining parameters in the input and output specifications as detailed in the sections below.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Setup MW REST base URL..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:21.811371700Z",
     "start_time": "2023-10-09T13:19:21.772879500Z"
    }
   },
   "outputs": [],
   "source": [
    "MWBaseURL = \"https://www.metabolomicsworkbench.org/rest\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The “study” context**\n",
    "\n",
    "The \"study\" context refers to the studies available in the Metabolomics Workbench (www.metabolomicsworkbench.org), a public repository for metabolomics metadata and experimental data spanning various species and experimental platforms, metabolite standards, metabolite structures, protocols, tutorials and training material, and other educational resources. It provides a computational platform to integrate, analyze, track, deposit, and disseminate large volumes of heterogeneous data from a wide variety of metabolomics studies including Mass Spectrometry (MS) and Nuclear Magnetic Resonance (NMR) spectrometry data spanning a variety of species covering all the major taxonomic categories including humans and other mammals, plants, insects, invertebrates, and microorganisms. This context provides access to a variety of data associated with studies such as study summary, experimental factors for study design, analysis information, metabolites and results data, sample source and species etc.\n",
    "\n",
    "context = study\n",
    "\n",
    "input_item = study_id | study_title | institute | last_name | analysis_id | metabolite_id\n",
    "\n",
    "input_value = input_item_value\n",
    "\n",
    "output_item = summary | factors | analysis | metabolites | mwtab | source | species | disease | number_of_metabolites | data | datatable | untarg_studies | untarg_factors | untarg_data\n",
    "\n",
    "output_format = txt | json (Default: json)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "**Retrieve and process study data for in JSON format**\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Setup REST URL to retrieve data for a study ID..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:22.056055500Z",
     "start_time": "2023-10-09T13:19:21.819370900Z"
    }
   },
   "outputs": [],
   "source": [
    "MWDataURL = MWBaseURL + \"/study/study_id/ST000001/summary/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Execute REST request using \"request\" module..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:22.784654500Z",
     "start_time": "2023-10-09T13:19:21.831937500Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initiating request: https://www.metabolomicsworkbench.org/rest/study/study_id/ST000001/summary/\n"
     ]
    }
   ],
   "source": [
    "print(\"Initiating request: %s\" % MWDataURL)\n",
    "\n",
    "Response = requests.get(MWDataURL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check \"request\" status..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:22.827202700Z",
     "start_time": "2023-10-09T13:19:22.804201100Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Status Code: 200\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nStatus Code: %d\" % (Response.status_code))\n",
    "\n",
    "if Response.status_code != 200:\n",
    "    print(\"Request failed: status_code: %d\" % Response.status_code)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Process JSON results..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:23.219652100Z",
     "start_time": "2023-10-09T13:19:22.905197400Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Available data for a study summary:\n",
      "\n",
      "study_id: ST000001\n",
      "study_title: Fatb Induction Experiment (FatBIE)\n",
      "study_type: Genotype treatment\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "phone: \n",
      "submit_date: 2013-01-15\n",
      "study_summary: This experiment tests the consequence of a mutation at the FatB gene (At1g08510) in the wound-response of Arabidopsis.  The FatB mutant allele (fatb KD J. Ohlrogge (Plant Cell 2003, Vol 15, 1020-1033)) was obtained from Dr. Katayonn Dehesh, University of California, Davis, Davis, CA.  This allele is in the Ws background.The standardized growth conditions are as follows: 1.  Seeds (between 14 and 16) are sown on media in 100 x 100 x 15mm square Falcon Petri Dishes (Fisher Scientific, catalogue #08-757-11A).  Seeds were arranged on the plates in a single horizontal line at the 1-cm mark from the top of the plate.2.  Each plate contains between 20 and 25-ml of sterile MS media containing 0.1% (w/v) sucrose.3.  Prior to sowing, seeds were sterilized by treating for 1 minute at room temperature with a 300-l solution of 50% (v/v) ethanol, this solution was removed and replaced with a 300-l solution consisting of 1% (v/v) Tween 20 (Fischer BioReagents, catalogue #BP33750), and 50% (v/v) bleach solution (Clorox), and incubated at room temperature for 10-minutes.  The seeds were then washed with three changes of 0.3-ml of sterile water.\n",
      "subject_species: Arabidopsis thaliana\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nAvailable data for a study summary:\\n\")\n",
    "\n",
    "Results = Response.json()\n",
    "\n",
    "for ResultType in Results:\n",
    "    ResultValue = Results[ResultType]\n",
    "    print(\"%s: %s\" % (ResultType, ResultValue))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "**Retrieve and process study data for multiple studies in JSON format**\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A substring may be specified as a study ID to retrieve data for multiple studies. The specified study ID substring is matched against all study IDs to retrieve data for matched studies. The study ID substring \"ST\" matches against all available studies.\n",
    "\n",
    "Setup REST URL to retrieve data for studies ST000010 to ST000019 by using ST00001 as study ID..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:23.632083100Z",
     "start_time": "2023-10-09T13:19:23.226652100Z"
    }
   },
   "outputs": [],
   "source": [
    "MWDataURL = MWBaseURL + \"/study/study_id/ST00001/summary/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Execute REST request and check the status..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:24.891904300Z",
     "start_time": "2023-10-09T13:19:23.776286900Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initiating request: https://www.metabolomicsworkbench.org/rest/study/study_id/ST00001/summary/\n",
      "\n",
      "Status Code: 200\n"
     ]
    }
   ],
   "source": [
    "print(\"Initiating request: %s\" % MWDataURL)\n",
    "\n",
    "Response = requests.get(MWDataURL)\n",
    "\n",
    "print(\"\\nStatus Code: %d\" % (Response.status_code))\n",
    "\n",
    "if Response.status_code != 200:\n",
    "    print(\"Request failed: status_code: %d\" % Response.status_code)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Process JSON results..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:24.942982100Z",
     "start_time": "2023-10-09T13:19:24.886919900Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Available data for studies:\n",
      "\n",
      "\n",
      "ResultNum: 1\n",
      "\n",
      "study_id: ST000010\n",
      "study_title: Lung Cancer Cells 4\n",
      "study_type: MS analysis (Untargeted)\n",
      "institute: University of Michigan\n",
      "department: \n",
      "last_name: Keshamouni\n",
      "first_name: Venkat\n",
      "email: vkeshamo@umich.edu\n",
      "phone: \n",
      "submit_date: 2013-04-03\n",
      "study_summary: In cancer cells, the process of epithelial–mesenchymal transition (EMT) confers migratory and invasive capacity, resistance to apoptosis, drug resistance, evasion of host immune surveillance and tumor stem cell traits. Cells undergoing EMT may represent tumor cells with metastatic potential. Characterizing the EMT secretome may identify biomarkers to monitor EMT in tumor progression and provide a prognostic signature to predict patient survival. Utilizing a transforming growth factor-β-induced cell culture model of EMT, we quantitatively profiled differentially secreted proteins, by GeLC-tandem mass spectrometry. Integrating with the corresponding transcriptome, we derived an EMT-associated secretory phenotype (EASP) comprising of proteins that were differentially upregulated both at protein and mRNA levels. Four independent primary tumor-derived gene expression data sets of lung cancers were used for survival analysis by the random survival forests (RSF) method. Analysis of 97-gene EASP expression in human lung adenocarcinoma tumors revealed strong positive correlations with lymph node metastasis, advanced tumor stage and histological grade. RSF analysis built on a training set (n = 442), including age, sex and stage as variables, stratified three independent lung cancer data sets into low-, medium- and high-risk groups with significant differences in overall survival. We further refined EASP to a 20 gene signature (rEASP) based on variable importance scores from RSF analysis. Similar to EASP, rEASP predicted survival of both adenocarcinoma and squamous carcinoma patients. More importantly, it predicted survival in the early-stage cancers. These results demonstrate that integrative analysis of the critical biological process of EMT provides mechanism-based and clinically relevant biomarkers with significant prognostic value.\n",
      "Research is published, core data not used but project description is relevant:\n",
      "http://www.jimmunol.org/content/194/12/5789.long\n",
      "\n",
      "subject_species: Homo sapiens\n",
      "\n",
      "ResultNum: 2\n",
      "\n",
      "study_id: ST000011\n",
      "study_title: African Metabolomics\n",
      "study_type: MS analysis\n",
      "institute: University of Michigan\n",
      "department: \n",
      "last_name: Thompson\n",
      "first_name: Simon\n",
      "email: simontho@upenn.edu\n",
      "phone: \n",
      "submit_date: 2013-04-04\n",
      "study_summary: Compare metabolomic, phenotypic and genetic diversity across various groups of Africans.\n",
      "subject_species: Homo sapiens\n",
      "\n",
      "ResultNum: 3\n",
      "\n",
      "study_id: ST000012\n",
      "study_title: Extraction Optimization Study\n",
      "study_type: Solvent treatment\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "phone: \n",
      "submit_date: 2013-05-20\n",
      "study_summary: Steel Ball/ Glass bead + Chloroform/ Isopropanol treatment\n",
      "subject_species: Chlamydomonas reinhardtii\n",
      "\n",
      "ResultNum: 4\n",
      "\n",
      "study_id: ST000013\n",
      "study_title: Mutation Study\n",
      "study_type: Genotype\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "phone: \n",
      "submit_date: 2013-05-20\n",
      "study_summary: This experiment tests the effect of individual mutations on the metabolome of Arabidopsis. The standardized growth conditions are as follows: 1. Seeds (between 14 and 16) are sown on media in 100 x 100 x 15mm square Falcon Petri Dishes (Fisher Scientific, catalogue #08-757-11A). Seeds were arranged on the plates in a single horizontal line at the 1-cm mark from the top of the plate. 2. Each plate contains between 20 and 25-ml of sterile MS media containing 0.1% (w/v) sucrose. 3. Prior to sowing, seeds were sterilized by treating for 1-minute at room temperature with a 300-l solution of 50% (v/v) ethanol, this solution was removed and replaced with a 300-l solution consisting of 1% (v/v) Tween 20 (Fischer BioReagents, catalogue #BP33750), and 50% (v/v) bleach solution (Clorox), and incubated at room temperature for 10-minutes. The seeds were then washed with three changes of 0.3-ml of sterile water.  4. Upon sowing with seeds, the plates were wrapped with Micropore tape (3M Health Care, catalogue #1530-0), and then stored horizontally for 4-days at 4 &deg;C, with illumination of 1 mol/m2. 5. On the 5th day, plates were moved to the growth room, and held in a vertical position in Plexi-glass holders for 17-days this growth room condition is labeled in Table I. 6. On 18th day Petri plates were opened and the aerial portions of these plants were harvested immediately upon plate opening. 7. Upon harvesting, plant material was quenched by immersion in liquid nitrogen and stored at 70 &deg;C.\n",
      "subject_species: Arabidopsis thaliana\n",
      "\n",
      "ResultNum: 5\n",
      "\n",
      "study_id: ST000014\n",
      "study_title: Skeletal Muscle Treatment (palmitate/palmitoyl carnitine)\n",
      "study_type: MS analysis\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "phone: \n",
      "submit_date: 2013-05-20\n",
      "study_summary: Isolated skelelal muscle mitochondria incubated in the presence of fatty acid oxidation precursors, palmitic acid and palmitoyl carnitine\n",
      "subject_species: Mus musculus\n",
      "\n",
      "ResultNum: 6\n",
      "\n",
      "study_id: ST000015\n",
      "study_title: Skeletal Muscle Treatment (palmitate and inhibitors)\n",
      "study_type: MS analysis\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "phone: \n",
      "submit_date: 2013-05-20\n",
      "study_summary: Isolated skelelal muscle mitochondria incubated in the presence of fatty acid oxidation precursor and inhibitors\n",
      "subject_species: Mus musculus\n",
      "\n",
      "ResultNum: 7\n",
      "\n",
      "study_id: ST000016\n",
      "study_title: NPM-ALK metabolic regulation\n",
      "study_type: LC-MS analysis (Untargeted)\n",
      "institute: University of Michigan\n",
      "department: Dept. of Pathology\n",
      "last_name: McDonnell\n",
      "first_name: Scott\n",
      "email: scottmcd@med.umich.edu\n",
      "phone: 734-936-1873\n",
      "submit_date: 2013-09-24\n",
      "study_summary: The mechanisms underlying the pathogenesis of the constitutively active tyrosine kinase nucleophosmin-anaplastic lymphoma kinase (NPM-ALK) expressing anaplastic large cell lymphoma are not completely understood. Here we show using an integrated phosphoproteomic and metabolomic strategy that NPM-ALK induces a metabolic shift toward aerobic glycolysis, increased lactate production, and biomass production. The metabolic shift is mediated through the anaplastic lymphoma kinase (ALK) phosphorylation of the tumor-specific isoform of pyruvate kinase (PKM2) at Y105, resulting in decreased enzymatic activity. Small molecule activation of PKM2 or expression of Y105F PKM2 mutant leads to reversal of the metabolic switch with increased oxidative phosphorylation and reduced lactate production coincident with increased cell death, decreased colony formation, and reduced tumor growth in an in vivo xenograft model. This study provides comprehensive profiling of the phosphoproteomic and metabolomic consequences of NPM-ALK expression and reveals a novel role of ALK in the regulation of multiple components of cellular metabolism. Our studies show that PKM2 is a novel substrate of ALK and plays a critical role in mediating the metabolic shift toward biomass production and tumorigenesis.\n",
      "Research is published:\n",
      "http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3739039/\n",
      "\n",
      "subject_species: Homo sapiens\n",
      "\n",
      "ResultNum: 8\n",
      "\n",
      "study_id: ST000017\n",
      "study_title: Rat HCR/LCR Stamina Study\n",
      "study_type: LC-MS analysis (Untargeted)\n",
      "institute: University of Michigan\n",
      "department: Internal Medicine\n",
      "last_name: Qi\n",
      "first_name: Nathan\n",
      "email: nathanqi@med.umich.edu\n",
      "phone: 734-232-0815\n",
      "submit_date: 2013-09-24\n",
      "study_summary: Maximal exercise-associated oxidative capacity is strongly correlated with health and longevity in humans. Rats selectively bred for high running capacity (HCR) have improved metabolic health and are longer-lived than their low-capacity counterparts (LCR). Using metabolomic and proteomic profiling, we show that HCR efficiently oxidize fatty acids (FAs) and branched-chain amino acids (BCAAs), sparing glycogen and reducing accumulation of short- and medium-chain acylcarnitines. HCR mitochondria have reduced acetylation of mitochondrial proteins within oxidative pathways at rest, and there is rapid protein deacetylation with exercise, which is greater in HCR than LCR. Fluxomic analysis of valine degradation with exercise demonstrates a functional role of differential protein acetylation in HCR and LCR. Our data suggest that efficient FA and BCAA utilization contribute to high intrinsic exercise capacity and the health and longevity benefits associated with enhanced fitness.\n",
      "Research is published, untargeted data not used in publication but project description is relevant:\n",
      "http://www.sciencedirect.com/science/article/pii/S1550413115000595\n",
      "\n",
      "subject_species: Rattus norvegicus\n",
      "\n",
      "ResultNum: 9\n",
      "\n",
      "study_id: ST000019\n",
      "study_title: Determine purity and quality of IROA labelled glucose\n",
      "study_type:  NMR and MS analysis\n",
      "institute: University of Florida\n",
      "department: Biochemistry & Molecular Biology\n",
      "last_name: Stupp\n",
      "first_name: Gregory\n",
      "email: stuppie@ufl.edu\n",
      "phone: \n",
      "submit_date: 2013-12-10\n",
      "study_summary: We demonstrate the global metabolic analysis ofCaenorhabditis elegansstress responses using a mass-spectrometry-based technique called isotopic ratio outlier analysis (IROA). In an IROA protocol, control and experimental samples are isotopically labeled with 95 and 5%13C, and the two sample populations are mixed together for uniform extraction, sample preparation, and LC-MS analysis.To illustrate the utility of IROA for global metabolomics, we exposed wild-type (N2) worms to a heat shock (30 min heat shock at 33 C), which causes significant, widespread changes in metabolism. We collected and analyzed material from the exometabolome (all material that worms release in the supernatant) and the endometabolome (homogenized total extracts from the worm bodies).\n",
      "subject_species: Caenorhabditis elegans\n",
      "\n",
      "Total number of studies matched: 9\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nAvailable data for studies:\\n\")\n",
    "\n",
    "Results = Response.json()\n",
    "\n",
    "StudiesCount = 0\n",
    "for ResultNum in Results:\n",
    "    StudiesCount += 1\n",
    "    print(\"\\nResultNum: %s\\n\" % ResultNum)\n",
    "\n",
    "    for ResultType in Results[ResultNum]:\n",
    "        ResultValue = Results[ResultNum][ResultType]\n",
    "        print(\"%s: %s\" % (ResultType, ResultValue))\n",
    "\n",
    "print(\"\\nTotal number of studies matched: %d\" % StudiesCount)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "**Retrieve and process study data in text format**\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set up REST URL..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:25.358328200Z",
     "start_time": "2023-10-09T13:19:24.929538500Z"
    }
   },
   "outputs": [],
   "source": [
    "MWDataURL = MWBaseURL + \"/study/study_id/ST000001/summary/txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Execute REST request using \"request\" module..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true,
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:27.204674500Z",
     "start_time": "2023-10-09T13:19:25.352325700Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initiating request: https://www.metabolomicsworkbench.org/rest/study/study_id/ST000001/summary/txt\n"
     ]
    }
   ],
   "source": [
    "print(\"Initiating request: %s\" % MWDataURL)\n",
    "\n",
    "Response = requests.get(MWDataURL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check \"request\" status..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:27.906782600Z",
     "start_time": "2023-10-09T13:19:27.195675800Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Status Code: 200\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nStatus Code: %d\" % (Response.status_code))\n",
    "\n",
    "if Response.status_code != 200:\n",
    "    print(\"Request failed: status_code: %d\" % Response.status_code)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Process text results..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:28.104691800Z",
     "start_time": "2023-10-09T13:19:27.903812800Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Available data for study summary:\n",
      "\n",
      "study_id: ST000001\n",
      "study_title: Fatb Induction Experiment (FatBIE)\n",
      "study_type: Genotype treatment\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "submit_date: 2013-01-15\n",
      "study_summary: This experiment tests the consequence of a mutation at the FatB gene (At1g08510) in the wound-response of Arabidopsis.  The FatB mutant allele (fatb KD J. Ohlrogge (Plant Cell 2003, Vol 15, 1020-1033)) was obtained from Dr. Katayonn Dehesh, University of California, Davis, Davis, CA.  This allele is in the Ws background.The standardized growth conditions are as follows: 1.  Seeds (between 14 and 16) are sown on media in 100 x 100 x 15mm square Falcon Petri Dishes (Fisher Scientific, catalogue #08-757-11A).  Seeds were arranged on the plates in a single horizontal line at the 1-cm mark from the top of the plate.2.  Each plate contains between 20 and 25-ml of sterile MS media containing 0.1% (w/v) sucrose.3.  Prior to sowing, seeds were sterilized by treating for 1 minute at room temperature with a 300-l solution of 50% (v/v) ethanol, this solution was removed and replaced with a 300-l solution consisting of 1% (v/v) Tween 20 (Fischer BioReagents, catalogue #BP33750), and 50% (v/v) bleach solution (Clorox), and incubated at room temperature for 10-minutes.  The seeds were then washed with three changes of 0.3-ml of sterile water.\n",
      "subject_species: Arabidopsis thaliana\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nAvailable data for study summary:\\n\")\n",
    "\n",
    "Results = Response.text\n",
    "for Result in Results.split(\"\\n\"):\n",
    "    Words = Result.split(\"\\t\")\n",
    "    if len(Words) != 2:\n",
    "        continue\n",
    "\n",
    "    ResultType, ResultValue = Result.split(\"\\t\")\n",
    "    print(\"%s: %s\" % (ResultType, ResultValue))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "**Retrieve and process data for multiple studies in text format**\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Setup REST URL to retrieve data for studies ST000010 to ST000019 by using ST00001 as study ID..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:28.361326300Z",
     "start_time": "2023-10-09T13:19:28.091694600Z"
    }
   },
   "outputs": [],
   "source": [
    "MWDataURL = MWBaseURL + \"/study/study_id/ST00001/summary/txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Execute REST request and check the status..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:29.624431Z",
     "start_time": "2023-10-09T13:19:28.360326300Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initiating request: https://www.metabolomicsworkbench.org/rest/study/study_id/ST00001/summary/txt\n",
      "\n",
      "Status Code: 200\n"
     ]
    }
   ],
   "source": [
    "print(\"Initiating request: %s\" % MWDataURL)\n",
    "\n",
    "Response = requests.get(MWDataURL)\n",
    "\n",
    "print(\"\\nStatus Code: %d\" % (Response.status_code))\n",
    "\n",
    "if Response.status_code != 200:\n",
    "    print(\"Request failed: status_code: %d\" % Response.status_code)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Process text results..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-10-09T13:19:29.629429900Z",
     "start_time": "2023-10-09T13:19:29.554890700Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Available summary data for studies:\n",
      "\n",
      "study_id: ST000010\n",
      "study_title: Lung Cancer Cells 4\n",
      "study_type: MS analysis (Untargeted)\n",
      "institute: University of Michigan\n",
      "last_name: Keshamouni\n",
      "first_name: Venkat\n",
      "email: vkeshamo@umich.edu\n",
      "submit_date: 2013-04-03\n",
      "study_summary: In cancer cells, the process of epithelial–mesenchymal transition (EMT) confers migratory and invasive capacity, resistance to apoptosis, drug resistance, evasion of host immune surveillance and tumor stem cell traits. Cells undergoing EMT may represent tumor cells with metastatic potential. Characterizing the EMT secretome may identify biomarkers to monitor EMT in tumor progression and provide a prognostic signature to predict patient survival. Utilizing a transforming growth factor-β-induced cell culture model of EMT, we quantitatively profiled differentially secreted proteins, by GeLC-tandem mass spectrometry. Integrating with the corresponding transcriptome, we derived an EMT-associated secretory phenotype (EASP) comprising of proteins that were differentially upregulated both at protein and mRNA levels. Four independent primary tumor-derived gene expression data sets of lung cancers were used for survival analysis by the random survival forests (RSF) method. Analysis of 97-gene EASP expression in human lung adenocarcinoma tumors revealed strong positive correlations with lymph node metastasis, advanced tumor stage and histological grade. RSF analysis built on a training set (n = 442), including age, sex and stage as variables, stratified three independent lung cancer data sets into low-, medium- and high-risk groups with significant differences in overall survival. We further refined EASP to a 20 gene signature (rEASP) based on variable importance scores from RSF analysis. Similar to EASP, rEASP predicted survival of both adenocarcinoma and squamous carcinoma patients. More importantly, it predicted survival in the early-stage cancers. These results demonstrate that integrative analysis of the critical biological process of EMT provides mechanism-based and clinically relevant biomarkers with significant prognostic value.\n",
      "\n",
      "\n",
      "\n",
      "subject_species: Homo sapiens\n",
      "\n",
      "study_id: ST000011\n",
      "study_title: African Metabolomics\n",
      "study_type: MS analysis\n",
      "institute: University of Michigan\n",
      "last_name: Thompson\n",
      "first_name: Simon\n",
      "email: simontho@upenn.edu\n",
      "submit_date: 2013-04-04\n",
      "study_summary: Compare metabolomic, phenotypic and genetic diversity across various groups of Africans.\n",
      "subject_species: Homo sapiens\n",
      "\n",
      "study_id: ST000012\n",
      "study_title: Extraction Optimization Study\n",
      "study_type: Solvent treatment\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "submit_date: 2013-05-20\n",
      "study_summary: Steel Ball/ Glass bead + Chloroform/ Isopropanol treatment\n",
      "subject_species: Chlamydomonas reinhardtii\n",
      "\n",
      "study_id: ST000013\n",
      "study_title: Mutation Study\n",
      "study_type: Genotype\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "submit_date: 2013-05-20\n",
      "study_summary: This experiment tests the effect of individual mutations on the metabolome of Arabidopsis. The standardized growth conditions are as follows: 1. Seeds (between 14 and 16) are sown on media in 100 x 100 x 15mm square Falcon Petri Dishes (Fisher Scientific, catalogue #08-757-11A). Seeds were arranged on the plates in a single horizontal line at the 1-cm mark from the top of the plate. 2. Each plate contains between 20 and 25-ml of sterile MS media containing 0.1% (w/v) sucrose. 3. Prior to sowing, seeds were sterilized by treating for 1-minute at room temperature with a 300-l solution of 50% (v/v) ethanol, this solution was removed and replaced with a 300-l solution consisting of 1% (v/v) Tween 20 (Fischer BioReagents, catalogue #BP33750), and 50% (v/v) bleach solution (Clorox), and incubated at room temperature for 10-minutes. The seeds were then washed with three changes of 0.3-ml of sterile water.  4. Upon sowing with seeds, the plates were wrapped with Micropore tape (3M Health Care, catalogue #1530-0), and then stored horizontally for 4-days at 4 &deg;C, with illumination of 1 mol/m2. 5. On the 5th day, plates were moved to the growth room, and held in a vertical position in Plexi-glass holders for 17-days this growth room condition is labeled in Table I. 6. On 18th day Petri plates were opened and the aerial portions of these plants were harvested immediately upon plate opening. 7. Upon harvesting, plant material was quenched by immersion in liquid nitrogen and stored at 70 &deg;C.\n",
      "subject_species: Arabidopsis thaliana\n",
      "\n",
      "study_id: ST000014\n",
      "study_title: Skeletal Muscle Treatment (palmitate/palmitoyl carnitine)\n",
      "study_type: MS analysis\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "submit_date: 2013-05-20\n",
      "study_summary: Isolated skelelal muscle mitochondria incubated in the presence of fatty acid oxidation precursors, palmitic acid and palmitoyl carnitine\n",
      "subject_species: Mus musculus\n",
      "\n",
      "study_id: ST000015\n",
      "study_title: Skeletal Muscle Treatment (palmitate and inhibitors)\n",
      "study_type: MS analysis\n",
      "institute: University of California, Davis\n",
      "department: Davis Genome Center\n",
      "last_name: Kind\n",
      "first_name: Tobias\n",
      "email: tkind@ucdavis.edu\n",
      "submit_date: 2013-05-20\n",
      "study_summary: Isolated skelelal muscle mitochondria incubated in the presence of fatty acid oxidation precursor and inhibitors\n",
      "subject_species: Mus musculus\n",
      "\n",
      "study_id: ST000016\n",
      "study_title: NPM-ALK metabolic regulation\n",
      "study_type: LC-MS analysis (Untargeted)\n",
      "institute: University of Michigan\n",
      "department: Dept. of Pathology\n",
      "last_name: McDonnell\n",
      "first_name: Scott\n",
      "email: scottmcd@med.umich.edu\n",
      "phone: 734-936-1873\n",
      "submit_date: 2013-09-24\n",
      "study_summary: The mechanisms underlying the pathogenesis of the constitutively active tyrosine kinase nucleophosmin-anaplastic lymphoma kinase (NPM-ALK) expressing anaplastic large cell lymphoma are not completely understood. Here we show using an integrated phosphoproteomic and metabolomic strategy that NPM-ALK induces a metabolic shift toward aerobic glycolysis, increased lactate production, and biomass production. The metabolic shift is mediated through the anaplastic lymphoma kinase (ALK) phosphorylation of the tumor-specific isoform of pyruvate kinase (PKM2) at Y105, resulting in decreased enzymatic activity. Small molecule activation of PKM2 or expression of Y105F PKM2 mutant leads to reversal of the metabolic switch with increased oxidative phosphorylation and reduced lactate production coincident with increased cell death, decreased colony formation, and reduced tumor growth in an in vivo xenograft model. This study provides comprehensive profiling of the phosphoproteomic and metabolomic consequences of NPM-ALK expression and reveals a novel role of ALK in the regulation of multiple components of cellular metabolism. Our studies show that PKM2 is a novel substrate of ALK and plays a critical role in mediating the metabolic shift toward biomass production and tumorigenesis.\n",
      "\n",
      "\n",
      "\n",
      "subject_species: Homo sapiens\n",
      "\n",
      "study_id: ST000017\n",
      "study_title: Rat HCR/LCR Stamina Study\n",
      "study_type: LC-MS analysis (Untargeted)\n",
      "institute: University of Michigan\n",
      "department: Internal Medicine\n",
      "last_name: Qi\n",
      "first_name: Nathan\n",
      "email: nathanqi@med.umich.edu\n",
      "phone: 734-232-0815\n",
      "submit_date: 2013-09-24\n",
      "study_summary: Maximal exercise-associated oxidative capacity is strongly correlated with health and longevity in humans. Rats selectively bred for high running capacity (HCR) have improved metabolic health and are longer-lived than their low-capacity counterparts (LCR). Using metabolomic and proteomic profiling, we show that HCR efficiently oxidize fatty acids (FAs) and branched-chain amino acids (BCAAs), sparing glycogen and reducing accumulation of short- and medium-chain acylcarnitines. HCR mitochondria have reduced acetylation of mitochondrial proteins within oxidative pathways at rest, and there is rapid protein deacetylation with exercise, which is greater in HCR than LCR. Fluxomic analysis of valine degradation with exercise demonstrates a functional role of differential protein acetylation in HCR and LCR. Our data suggest that efficient FA and BCAA utilization contribute to high intrinsic exercise capacity and the health and longevity benefits associated with enhanced fitness.\n",
      "\n",
      "\n",
      "\n",
      "subject_species: Rattus norvegicus\n",
      "\n",
      "study_id: ST000019\n",
      "study_title: Determine purity and quality of IROA labelled glucose\n",
      "study_type:  NMR and MS analysis\n",
      "institute: University of Florida\n",
      "department: Biochemistry & Molecular Biology\n",
      "last_name: Stupp\n",
      "first_name: Gregory\n",
      "email: stuppie@ufl.edu\n",
      "submit_date: 2013-12-10\n",
      "study_summary: We demonstrate the global metabolic analysis ofCaenorhabditis elegansstress responses using a mass-spectrometry-based technique called isotopic ratio outlier analysis (IROA). In an IROA protocol, control and experimental samples are isotopically labeled with 95 and 5%13C, and the two sample populations are mixed together for uniform extraction, sample preparation, and LC-MS analysis.To illustrate the utility of IROA for global metabolomics, we exposed wild-type (N2) worms to a heat shock (30 min heat shock at 33 C), which causes significant, widespread changes in metabolism. We collected and analyzed material from the exometabolome (all material that worms release in the supernatant) and the endometabolome (homogenized total extracts from the worm bodies).\n",
      "subject_species: Caenorhabditis elegans\n",
      "\n",
      "\n",
      "\n",
      "Total number of studies matched: 9\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nAvailable summary data for studies:\\n\")\n",
    "\n",
    "Results = Response.text\n",
    "\n",
    "StudiesCount = 0\n",
    "for Result in Results.split(\"\\n\"):\n",
    "    Words = Result.split(\"\\t\")\n",
    "    if len(Words) != 2:\n",
    "        print(\"\")\n",
    "        continue\n",
    "\n",
    "    ResultType, ResultValue = Words\n",
    "    if re.match(\"^study_id$\", ResultType, re.I):\n",
    "        StudiesCount += 1\n",
    "    print(\"%s: %s\" % (ResultType, ResultValue))\n",
    "\n",
    "print(\"\\nTotal number of studies matched: %d\" % StudiesCount)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
